src/Stream/ReadableStreamTrait.php

```diff
@@ -99,9 +99,7 @@ protected function performRead($chunkSize)
             if ($this->readCursor < $this->size) {
                 $this->readChunk();
             } else {
-                $this->emit('end', [
-                    $this,
-                ]);
+                $this->close();
```

Comment:
I'm not sure I follow exactly what' supposed to happen here, but it's my understanding that this should first emit an `end` event immediately followed by a `close`? (https://github.com/reactphp/stream#end-event)

Do we have a test for this?

Reply:
Good point, end shouldn't be emitted from the close call. But from when we reach EOF. Adjusted code and tests to ensure the behavior is as described on https://github.com/reactphp/stream#end-event

Reply:
Thank you for looking into this, changes LGTM! :+1: 

For the reference: The additional `$this` arguments have been removed via https://github.com/reactphp/stream/pull/69 a while back. This component still targets legacy stream v0.4, so I guess we may want to address this eventually.